File: admin/src/api/setting/website.ts
import request from '@/utils/request'

// 版权信息参数
export type type_setting_copyright_item = {
    name: string
    link: string
}

// 版权信息返回信息
export type type_setting_copyright_resp = type_setting_copyright_item[]

// 网站设置参数
export type type_setting_website_save = {
    name?: string
    logo?: string
    favicon?: string
    backdrop?: string
    shopName?: string
    shopLogo?: string
}

// 网站设置返回信息
export type type_setting_website_resp = {
    name: string
    logo: string
    favicon: string
    backdrop: string
    shopName: string
    shopLogo: string
}

// 获取备案信息
export function getCopyright() {
    return request.get<type_setting_copyright_resp>({ url: '/setting/copyright/detail' })
}

// 设置备案信息
export function setCopyright(data: type_setting_copyright_item[]) {
    return request.post({ url: '/setting/copyright/save', data })
}

// 获取网站信息
export function getWebsite() {
    return request.get<type_setting_website_resp>({ url: '/setting/website/detail' })
}

// 设置网站信息
export function setWebsite(data: type_setting_website_save) {
    return request.post({ url: '/setting/website/save', data })
}

// 获取网站接口列表
export function getApiList() {
    return request.get({ url: '/apiList' })
}

File: server/app/controller/admin_ctl/setting_controller/website.go
package setting_controller

import (
	"x_admin/app/schema/setting_schema"
	"x_admin/app/service/setting_service"
	"x_admin/core/response"
	"x_admin/util"

	"github.com/gin-gonic/gin"
)

// WebsiteHandler 网站设置控制器
type WebsiteHandler struct{}

// @Summary		获取网站信息
// @Description	获取网站设置信息
// @Tags			setting_website-网站设置
// @Param			token	header		string										true	"token"
// @Success		200		{object}	response.Response{data=map[string]string}	"成功"
// @Router			/api/admin/setting/website/detail [get]
func (wh WebsiteHandler) Detail(c *gin.Context) {
	res, err := setting_service.WebsiteService.Detail()
	response.CheckAndRespWithData(c, res, err)
}

// @Summary		保存网站信息
// @Description	保存网站设置信息
// @Tags			setting_website-网站设置
// @Param			token		header		string				true	"token"
// @Param			name		body		string				false	"网站名称"
// @Param			logo		body		string				false	"网站图标"
// @Param			favicon		body		string				false	"网站LOGO"
// @Param			backdrop	body		string				false	"登录页广告图"
// @Param			shopName	body		string				false	"商城名称"
// @Param			shopLogo	body		string				false	"商城Logo"
// @Success		200			{object}	response.Response	"成功"
// @Router			/api/admin/setting/website/save [post]
func (wh WebsiteHandler) Save(c *gin.Context) {
	var wsReq setting_schema.SettingWebsiteReq
	if response.IsFailWithResp(c, util.VerifyUtil.VerifyJSON(c, &wsReq)) {
		return
	}
	response.CheckAndRespWithData(c, nil, setting_service.WebsiteService.Save(wsReq))
}
